### ISSUE TITLE
[MSOURCES-27] Specifying excludes in the configuration removes the default excludes

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-source-plugin#107]
**[Daniel Kulp](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dkulp)** opened **[MSOURCES-27](https://issues.apache.org/jira/browse/MSOURCES-27?redirect=false)** and commented

source:jar with 2.0.3 is adding the .svn directories into the jar.    2.0.2 did not.

---

**Affects:** 2.0.3

**Comment by jira-importer** [comment]
**[Daniel Kulp](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dkulp)** commented

This one is a bit more complex.    For CXF, we have properties files in the src/main/java dir (i18n message files) that are included via resource.   Thus, in the pom, we have:

    <resources>
        <resource>
            <directory>src/main/java</directory>
            <excludes>
                <exclude>**/*.java</exclude>
            </excludes>
        </resource>
    </resources>

to grab all the non-java files as resources.

The new source plugin uses the **/*.java as an exclude, but but DOESN'T add the list of default excludes.   Thus, the source plugin will get the .svn dirs.

Basically, for resources that don't have excludes, the source plugin works fine as it uses the default excludes.   For those that DO define an exclude, it doesn't quite work as expected.

**Comment by jira-importer** [comment]
**[Daniel Kulp](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dkulp)** commented

Just checked the resource plugin.   The resource plugin does:

```
            if ( resource.getExcludes() != null && !resource.getExcludes().isEmpty() )
            {
                scanner.setExcludes( (String[]) resource.getExcludes().toArray( EMPTY_STRING_ARRAY ) );
            }

            scanner.addDefaultExcludes();
            scanner.scan();
```

Thus, it always adds the default excludes to the scanner that it uses.    The sources plugin should follow the same behavior to make sure the two are consistent.

**Comment by jira-importer** [comment]
**[Dennis Lundberg](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dennisl@apache.org)** commented

Do you expect the \<exclude> to mean: "use the default excludes and add this one"? And that was the behavior in maven-sources-plugin 2.0.2?

Just out of curiosity, is there a reason for not having the properties files in /src/main/resources/... ?

**Comment by jira-importer** [comment]
**[Daniel Kulp](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dkulp)** commented

Yep.  That looks right to me.   That matches 2.0.2 as well as the behavior of the resource plugin.
